### PR TITLE
feat(output): silence metric notice when --quiet is specified

### LINKED ISSUE #4104: keep --quiet output silent
**Describe the bug**
Some non-core output, including the metrics notice, is printed even with `--quiet`. We should add a test to make sure that the --quiet flag supresses everything except findings ouptut. 

**To Reproduce**
Run a fresh semgrep install with the `--quiet` flag.

**Expected behavior**
Only findings output is reported. If there are no findings, there is no output.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

### PR DESCRIPTION
**Pull request by brendongo**
Closes https://github.com/returntocorp/semgrep/issues/4104

PR checklist:
- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by emjin** [comment]
Nice :D We should add a test for this though

**Comment by ievans** [comment]
thx @brendongo; copy the tests from https://github.com/returntocorp/semgrep/pull/4105 probably?

**Comment by brendongo** [comment]
Yep and gonna copy in your changes to conftest

File: semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
         "metadata": {
           "category": "correctness",
           "license": "Commons Clause License Condition v1.0[LGPL-2.1-only]",
+          "shortlink": "https://sg.run/Kl6n",
```
Comment (brendongo) [inline semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json]:
```suggestion
```

Comment (emjin) [reply semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json]: I'm updating the tests: https://github.com/returntocorp/semgrep/pull/4106

Comment (emjin) [reply semgrep/tests/e2e/snapshots/test_check/test_registry_rule/results.json]: Will merge into develop soon

File: semgrep/tests/e2e/snapshots/test_check/test_terminal_output/error.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+METRICS: Using configs from the Registry (like --config=p/ci) reports pseudonymous rule metrics to semgrep.dev.
```
Comment (brendongo) [inline semgrep/tests/e2e/snapshots/test_check/test_terminal_output/error.txt:1]: This is the test that actually checks metrics are printed out

File: semgrep/tests/e2e/snapshots/test_check/test_terminal_output_quiet/output.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+targets/basic/stupid.js
```
Comment (brendongo) [inline semgrep/tests/e2e/snapshots/test_check/test_terminal_output_quiet/output.txt:1]: This is the test that checks --quiet doesnt print out metric message

File: semgrep/tests/e2e/snapshots/test_exclude_include/test_exclude_include/exclude-and-include/err.out [hunk 4]
```diff
@@ -1,4 +1,10 @@
 Paths that match both --include and --exclude will be skipped by Semgrep.
```
Comment (emjin) [inline semgrep/tests/e2e/snapshots/test_exclude_include/test_exclude_include/exclude-and-include/err.out]: lmao this output looks really dumb---in particular, having the metrics message between the paths messaging. Not important for this PR though

File: semgrep/tests/e2e/snapshots/test_check/test_terminal_output/error.txt (hunk 2 above)
Comment (emjin) [reply semgrep/tests/e2e/snapshots/test_check/test_terminal_output/error.txt:1]: Should we have a test that checks that metrics are only printed out on the first run?

**Review by emjin — APPROVED** [review]

File: semgrep/semgrep/cli.py [hunk 5]
```diff
@@ -580,6 +578,8 @@ def cli(
         verbose=verbose, debug=debug, quiet=quiet, force_color=force_color
     )
 
+    possibly_notify_user()
```
Comment (emjin) [inline semgrep/semgrep/cli.py:582]: Add a comment explaining that the order here is important

Comment (brendongo) [reply semgrep/semgrep/cli.py:582]: Good call. Temporal Coupling is usually a poor design smell and we should probably fix this at a later date :D 

Relevant reading: https://www.pluralsight.com/tech-blog/forms-of-temporal-coupling/

File: semgrep/semgrep/cli.py [hunk 6]
```diff
@@ -578,6 +578,7 @@ def cli(
         verbose=verbose, debug=debug, quiet=quiet, force_color=force_color
     )
 
+    # Note this should be after flags set to verbosity is respected
```
Comment (emjin) [inline semgrep/semgrep/cli.py]:
```suggestion
    # Note this must be after the call to `set_flags` so that verbosity is respected
```

**Pull request merged**